.github/workflows/ci.yml

```diff
@@ -101,7 +103,7 @@ jobs:
       - name: Display version
         run: check/pylint --version
       - name: Lint
-        run: check/pylint -v
+        run: check/pylint-changed-files
```

Comment:
I don't know if this is an acceptable change or not. This isn't just making the doc-linting incremental, but the whole run of pylint. We've already gone through the trouble of making the whole codebase pass pylint in full. In an ideal universe, the incremental version will make sure no lint gets added, but I'm always wary with running these sorts of tools on a subset of files as it may miss some issues that only crop up when you can see the full project. On the other hand: pylint itself takes a long time on the full repo, so this could help build times.

What do you think @balopat also cc @MichaelBroughton

Reply:
I don't think this is acceptable. If you change types/API in one file, that can easily break others that depend on it but didn't change. 

I would recommend creating a new job running on changed files only with a separate pylintrc that has the docs config.

Reply:
I'm not a huge fan of giving up full coverage either. @mpharrigan, @balopat  regarding slow pylint performance, had we ever tried just doing something like: `check/pylint -j 0 ` which uses all your machines cores for pylint instead of just the one ?

Reply:
I experimented with parallelization a while ago for pytest, the speed up is not that great on Github actions given their machines are 2vCPU (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners). Also, specifically this PR will have an issue: as the doc linter is a plugin and [pylint does not support](http://pylint.pycqa.org/en/latest/user_guide/run.html#parallel-execution) parallization with plugins :(

Reply:
OK, makes sense (although I always thought the concern of the linter is more about style and not about interdependencies).  I implemented the suggested change.